test/testpushnotifications.cpp

```diff
@@ -0,0 +1,279 @@
+        auto credentials = new CredentialsStub(user, password);
+        account->setCredentials(credentials);
+        QSignalSpy readySpy(account->pushNotifications(), &OCC::PushNotifications::ready);
+        QVERIFY(readySpy.isValid());
+
+        // Wait for authentication
+        QVERIFY(processTextMessageSpy.wait());
+
+        // Right authentication data should be sent
+        QCOMPARE(processTextMessageSpy.count(), 2);
+
+        const auto socket = processTextMessageSpy.at(0).at(0).value<QWebSocket *>();
+        const auto userSent = processTextMessageSpy.at(0).at(1).toString();
+        const auto passwordSent = processTextMessageSpy.at(1).at(1).toString();
+
+        QCOMPARE(userSent, user);
+        QCOMPARE(passwordSent, password);
+
+        // Sent authenticated
+        socket->sendTextMessage("authenticated");
+
+        // Wait for ready signal
+        readySpy.wait();
+        QCOMPARE(readySpy.count(), 1);
+        QCOMPARE(account->pushNotifications()->isReady(), true);
+    }
+
+    void testOnWebSocketTextMessageReceived_notifyFileMessage_emitFilesChanged()
+    {
+        const QString user = "user";
+        const QString password = "password";
+        FakeWebSocketServer fakeServer;
+        QSignalSpy processTextMessageSpy(&fakeServer, &FakeWebSocketServer::processTextMessage);
+        QVERIFY(processTextMessageSpy.isValid());
+
+        auto account = FakeWebSocketServer::createAccount();
+        auto credentials = new CredentialsStub(user, password);
+        account->setCredentials(credentials);
+        QSignalSpy filesChangedSpy(account->pushNotifications(), &OCC::PushNotifications::filesChanged);
+        QVERIFY(filesChangedSpy.isValid());
+
+        // Wait for authentication and then send notify_file push notification
+        QVERIFY(processTextMessageSpy.wait());
+        QCOMPARE(processTextMessageSpy.count(), 2);
+        const auto socket = processTextMessageSpy.at(0).at(0).value<QWebSocket *>();
+        socket->sendTextMessage("notify_file");
+
+        // filesChanged signal should be emitted
+        QVERIFY(filesChangedSpy.wait());
+        QCOMPARE(filesChangedSpy.count(), 1);
+        auto accountFilesChanged = filesChangedSpy.at(0).at(0).value<OCC::Account *>();
+        QCOMPARE(accountFilesChanged, account.data());
+    }
+
+    void testOnWebSocketTextMessageReceived_notifyActivityMessage_emitNotification()
+    {
+        const QString user = "user";
+        const QString password = "password";
+        FakeWebSocketServer fakeServer;
+        QSignalSpy processTextMessageSpy(&fakeServer, &FakeWebSocketServer::processTextMessage);
+        QVERIFY(processTextMessageSpy.isValid());
+
+        auto account = FakeWebSocketServer::createAccount();
+        auto credentials = new CredentialsStub(user, password);
+        account->setCredentials(credentials);
+        QSignalSpy filesChangedSpy(account->pushNotifications(), &OCC::PushNotifications::notification);
```

Comment:
Should be notificationSpy now I guess :-)

Reply:
right:)